### DISCUSSION TITLE
How implement language server with the worker api instead of Nodejs environment?

### DISCUSSION THREAD

**Discussion by abdalla-rko** [TypeFox/monaco-languageclient#349] [Q&A] [answered]
I'm trying to implement a language server without using a server (Nodejs environment), like vscode web.

**Comment by CGNonofr** [comment, marked as answer]
see https://code.visualstudio.com/api/extension-guides/web-extensions#language-server-protocol-in-web-extensions

Your LSP implementation needs to be written in javascript and run inside a worker, then it'll communicate through `sendMessage`

I don't think there is many available implementations for the moment, which language server do you want to use?

**Reply by abdalla-rko** [reply]
I want to make use of the lua-language-server by Sumneko. See https://github.com/sumneko/lua-language-server.

**Reply by CGNonofr** [reply]
This language server is written in Lua, I don't think you'll be able to run it in the browser. If you manage to do it, I'm interested!

**Reply by CGNonofr** [reply]
Is it available in vscode web?

**Reply by abdalla-rko** [reply]
No unfortunately  it's not supported in vscode web. I already spent some time it seems like it's not doable.

**Reply by abdalla-rko** [reply]
Another question, so if I want to make a custom language server and run it inside a worker, will this library support the communication?

**Reply by CGNonofr** [reply]
This library is protocol agnostic, it just allows to start a vscode languageclient over the monaco editor.

The communication over websocket is handled by monaco-jsonrpc, and you can replace it by anything you want pretty easily.

Td;dr: yes
